tests/test_reader.py

```diff
@@ -195,3 +195,23 @@ def _common_entries(*dcts):
         return
     for i in set(dcts[0]).intersection(*dcts[1:]):
         yield tuple(d[i] for d in dcts)
+
+
+def test_pickle(single_nd2):
+    import pickle
+
+    f = ND2File(single_nd2)
+    pf = pickle.dumps(f)
+    assert isinstance(pf, bytes)
+    f2: ND2File = pickle.loads(pf)
+    np.testing.assert_array_equal(f, f2)
+
+    d = f.to_dask()
+    pd = pickle.dumps(d)
+    assert isinstance(pd, bytes)
+    # FIXME: this line currently leaks a file handle...
```

Comment:
Is this due to the `__setstate__` method when unpickling?
`__setstate__` will open the file, but I don't see the mechanism (context manager?) that would close the file again for an unpickled array.  Maybe through garbage collection?